src/main/java/org/idpass/lite/IDPASSMap.java

```diff
@@ -0,0 +1,201 @@
+        this.lang = lang;
+        this.UIN = UIN;
+        this.gender = gender;
+        this.givenName = givenName;
+        this.surName = surName;
+        this.placeOfBirth = placeOfBirth;
+        this.dateOfBirth = dateOfBirth;
+        this.address = address;
+    }
+
+    public List<String> getLang() {
+        return lang;
+    }
+
+    public void setLang(List<String> lang) {
+        this.lang = lang;
+    }
+
+    public FieldDesc getUIN() {
+        return UIN;
+    }
+
+    public void setUIN(FieldDesc UIN) {
+        this.UIN = UIN;
+    }
+
+    public FieldDesc getGender() {
+        return gender;
+    }
+
+    public void setGender(FieldDesc gender) {
+        this.gender = gender;
+    }
+
+    public FieldDesc getGivenName() {
+        currentField = "givenName";
+        return givenName;
+    }
+
+    public void setGivenName(FieldDesc givenName) {
+        this.givenName = givenName;
+    }
+
+    public FieldDesc getSurName() {
+        currentField = "surName";
+        return surName;
+    }
+
+    public void setSurName(FieldDesc surName) {
+        this.surName = surName;
+    }
+
+    public FieldDesc getPlaceOfBirth() {
+        return placeOfBirth;
+    }
+
+    public void setPlaceOfBirth(FieldDesc placeOfBirth) {
+        this.placeOfBirth = placeOfBirth;
+    }
+
+    public FieldDesc getDateOfBirth() {
+        currentField = "dateOfBirth";
+        return dateOfBirth;
+    }
+
+    public void setDateOfBirth(FieldDesc dateOfBirth) {
+        this.dateOfBirth = dateOfBirth;
+    }
+
+    public FieldDesc getAddress() {
+        return address;
+    }
+
+    public void setAddress(FieldDesc address) {
+        this.address = address;
+    }
+
+    /**
+     * For most fields which are directly mapped such as "gender", the get() method is just
+     * a pass-through. But for fields with no direct mapping, such as the surName field, this method
+     * parses the input data argument which might contain "Doe, John"
+     * and extracts the "Doe" as the surName. The purpose of this method
+     * is to construct a surName from a fullName.
+     * @param data Source field value. For example, fullName = "Doe, John"
+     * @return Returns the best-effort attempt to extract the field value
+     */
+
+    public String get(String data) {
+        String ret = "";
+        if (data == null) return ret;
+        else ret = data;
+
+        String arr[];
+
+        try {
+            switch (currentField) {
+                case "surName":
+                    arr = ret.split("\\s*(,|\\s)\\s*");
+                    if (ret.contains(",")) {
+                        ret = arr[0];
+                    } else {
+                        List<String> L = Arrays.asList(arr);
+                        ret = L.get(L.size() - 1);
+                    }
+                   break;
+
+                case "givenName":

```

Comment:
This would need to be defined in the config mapper instead of being hardcoded because it will depend a lot on the country. Now that we have a full-name field, we can should this.